### PR TITLE
List Shopcarts API

### PR DESCRIPTION
**Pull request by LillisLuo**
Complete the route logic of Update Shopcart API in routes.py
Wrote seven test cases in test.models.py
Wrote eleven test cases in test.routes.py
截屏2023-07-05 14 59 32
截屏2023-07-04 21 14 04
截屏2023-07-04 21 14 27

### CODE REVIEW

File: service/routes.py [hunk 1]
```diff
@@ -68,8 +68,18 @@ def is_expected_content_type(expected_content_type):
 
 @app.route("/shopcarts", methods=["GET"])
 def list_shopcarts():
-    # TODO: implement list API logic
-    return make_response(jsonify([]), status.HTTP_200_OK)
+    app.logger.info("Request for shopcart list")
+    shopcarts = []
+
+    name = request.args.get("name")
```
Comment (wl2484) [inline service/routes.py:74]: lol 👍🏻

File: tests/test_routes.py [hunk 2]
```diff
@@ -190,6 +190,46 @@ def test_list_shopcart_items_none_items_error(self, mock_shopcart_get_by_id):
...
+        self._create_a_shopcart_with_items(2)
+        self._create_a_shopcart_with_items(3)
+        self._create_an_empty_shopcart(5)
+        resp = self.client.get(f"{BASE_URL}")
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        self.assertEqual(type(shopcart_a), Shopcart)
+        data = resp.get_json()
+        self.assertEqual(len(data), Shopcart.find_id())
+
+    def test_list_shopcart_of_a_customer(self):
```
Comment (wl2484) [inline tests/test_routes.py:220]: 💯

File: tests/test_routes.py [hunk 3]
```diff
@@ -635,3 +675,67 @@ def test_delete_invalid_shopcart(self):
         """It Should Delete an non-existing shopcart"""
         res = self.client.delete(f"{BASE_URL}/{NONEXIST_SHOPCART_ID}")
         self.assertEqual(res.status_code, status.HTTP_204_NO_CONTENT)
+
+    def test_unsupported_method_on_shopcart(self):
```
Comment (wl2484) [inline tests/test_routes.py:679]: thank you for adding this!! 💯

**Review by wl2484 — APPROVED** [review]
this is the medal for you 🏅👍🏻

**Review by momo100110 — APPROVED** [review]
LGTM

**Pull request merged**